src/pages/ShoppingCart.js

```diff
@@ -0,0 +1,142 @@
+import {faTrashCan} from '@fortawesome/free-solid-svg-icons';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import {BiMinus, BiPlus} from 'react-icons/bi';
+import {ToastContainer} from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import styled from 'styled-components';
+
+import useStore from '../hooks/useStore.js';
+import {displayToast} from '../utils/utils.js';
+
+export default function ShoppingCart() {
+  const {shoppingItems, removeShoppingItem, decreaseItemQuantity, increaseItemQuantity} = useStore(state => ({
+    shoppingItems: state.shoppingItems,
+    removeShoppingItem: state.removeShoppingItem,
+    decreaseItemQuantity: state.decreaseItemQuantity,
+    increaseItemQuantity: state.increaseItemQuantity,
+  }));
+
+  let toastId = null;
+
+  function subtotal() {
+    return shoppingItems
+      .map(item => item.price * item.quantity)
+      .reduce((previousValue, currentValue) => previousValue + currentValue, 0);
+  }
+
+  function handleDecreaseQuality(id) {
+    const findItem = shoppingItems.find(item => item.id === id);
+    if (findItem) {
+      findItem.quantity <= 1 ? displayToast(toastId, "Quantity can't be less than 1!") : decreaseItemQuantity(id);
```

Comment:
Same here with the `toastId`. Not sure if this is the best solution to set `let toadtId = null;`.